### ISSUE TITLE
Difficulty Implementing External Service Message Transmission and Enabling Network Participants to Edit Messages in Artery

### ISSUE THREAD

**Issue by il3241ya** [riebl/artery#327]
Hi!

I hope this message finds you well. I am currently working on a project where we need to implement the transmission of messages in the Artery using Protocol Buffers. The goal is to ensure that these messages can be propagated among network participants. While I've consulted the documentation, I couldn't find specific information on how to implement message transmission in Artery from a service external to SUMO.

I would greatly appreciate any assistance or guidance you can provide on addressing this issue. If there are any specific resources or examples available, it would be incredibly helpful.

Additionally, I have a question regarding granting each network participant the ability to edit these messages. Could you please provide insights or direct me to relevant documentation on this matter?

Thank you in advance for your time and support.

**Comment by riebl** [comment]
Hi @il3241ya,

you might be interested in the [transfusion](https://github.com/riebl/artery/tree/master/scenarios/transfusion) scenario which allows you to inject V2X data from outside and vice versa.

I am not sure if I have understood your question about editing messages, though. Any receiving entity can do whatever it wants with received data, so what limitation are you trying to solve?

SUMO has been the first and for quite some time the only mobility simulator supported by Artery. Today, we also have an implementation for OpenTrafficSim (OTS). So I am pretty confident that adding yet another simulator "moving vehicles around" is possible.

**Comment by riebl** [comment]
As you have correctly stated, the `TransfusionService` tries to connect to an already listening TCP socket at 127.0.0.1:33080 by default. I have just verified with the *socat* utility that it is working:
- run `socat STDIO TCP-LISTEN:33080` in a terminal
- start the "General" configuration of the *transfusion* scenario (run_transfusion)
I then see data arriving on the terminal as expected and also my "garbage" typed in the terminal is received by the `TransfusionService`. Does this work on your system too?

**Comment by il3241ya** [comment]
Yes, it works with the Transfusion scenario (run_transfusion)

**Comment by riebl** [comment]
I don't think that Artery is misconfigured but that the external application is not listening properly. Maybe it is opening the listening socket too late, i.e. after TransfusionService tries to connect?

**Comment by il3241ya** [comment]
Thank you very much for the answers! I found the right solution for our project, without using the transfusion service
